evaluator/src/main/kotlin/DependencyRule.kt

```diff
@@ -72,6 +72,11 @@ class DependencyRule(
     override fun issueSource() =
         "$name - ${pkg.id.toCoordinates()} (dependency of ${project.id.toCoordinates()} in scope $scopeName)"
 
+    /**
+     * Return the direct dependencies of the [dependency].
+     */
+    fun dependencies() = dependency.visitDependencies { it.toList() }
```

Comment:
I don't understand how this is supposed to be used.
There is already the matcher `DependencyRule.isAtTreeLevel` which can be used to limit a rule to 
a certain tree level, e.g. direct dependencies. How is the use case different from that one?

Reply:
> How is the targeted use case different from that one?

I don't want to iterate over all dependencies and filter those at level 0 only, but straight away only get the direct dependencies for efficiency.

> Yes, could you provide an example use case for a rule that would use this?

This commit was extracted from https://github.com/oss-review-toolkit/ort/pull/4977 where also an example use is given [here](https://github.com/oss-review-toolkit/ort/pull/4977/commits/12ebf5523f6a3ab285797a0d6e512a004ac82f32#diff-826eafbbb8a3949e640571ae685f52f399c36e9b39e552b45b6386305e57401aR86-R87).

> Also this should be called `directDependencies()` to be clear.

I'll do the renaming.

Reply:
> > Yes, could you provide an example use case for a rule that would use this?
> 
> This commit was extracted from #4977 where also an example use is given [here](https://github.com/oss-review-toolkit/ort/pull/4977/commits/12ebf5523f6a3ab285797a0d6e512a004ac82f32#diff-826eafbbb8a3949e640571ae685f52f399c36e9b39e552b45b6386305e57401aR86-R87).

So the main point is: I don't want to check for the dependency I'm iterating on in a `DependencyRule` at which level *it* is, but get *its* direct dependencies.